lib/resty/acme/client.lua

```diff
@@ -301,7 +305,12 @@ function _M:post(url, payload, headers)
     body = json.decode(resp.body)
   elseif resp.headers['Content-Type']:sub(1, 24) == "application/problem+json" then
     body = json.decode(resp.body)
-    return nil, nil, body.detail or body.type
+    if body.type == 'urn:ietf:params:acme:error:badNonce' then
```

Comment:
let's also check on server is indeed returning the `resp.headers["Replay-Nonce"]` header, otherwise we should just error out.

also let's make log a ngx_WARN level, as user should have visibilty of the behaviour